File: backend/dotnet-webapi/src/WorldCup.Api/Dtos/BestValueResultDto.cs
namespace WorldCup.Api.Dtos;

public class BestValueResultDto
{
    public bool WithinBudget { get; set; }
    public List<MatchWithCityDto> Matches { get; set; } = new();
    public OptimisedRouteDto? Route { get; set; }
    public CostBreakdownDto CostBreakdown { get; set; } = new();
    public List<string> CountriesVisited { get; set; } = new();
    public int MatchCount { get; set; }
    public string Message { get; set; } = string.Empty;
}

File: backend/java-spring/src/main/java/com/unosquare/worldcup/dto/BestValueResultDTO.java
package com.unosquare.worldcup.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BestValueResultDTO {
    private Boolean withinBudget;
    private List<MatchWithCityDTO> matches;
    private OptimisedRouteDTO route;
    private CostBreakdownDTO costBreakdown;
    private List<String> countriesVisited;
    private Integer matchCount;
    private String message;
}
